tests/test_pathology_transforms.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from parameterized import parameterized
+
+from monai.apps.pathology.transforms import ExtractStainsMacenko
+from monai.utils import exact_version, optional_import
+
+cp, has_cp = optional_import("cupy", "8.6.0", exact_version)
+
+# input pixels all transparent and below the beta absorbance threshold
+EXTRACT_STAINS_TEST_CASE_1 = [
+    cp.full((3, 2, 3), 240),
```

Comment:
Calling cupy outside of the testing class scope is troublesome when it is not available. You should put all of them into a function and call it from within the test class. You can find an example here:
https://github.com/Project-MONAI/MONAI/blob/e0035b108899e70084b988b6bb78d707313f493d/tests/test_lesion_froc.py#L305-L306